src/shared/pkcs11-util.c

```diff
@@ -214,10 +214,14 @@ int pkcs11_token_login(
                 _cleanup_free_ char *text = NULL;
                 char **i, *e;
 
-                if (FLAGS_SET(token_info->flags, CKF_USER_PIN_FINAL_TRY))
+                if (FLAGS_SET(token_info->flags, CKF_USER_PIN_FINAL_TRY)) {
                         r = asprintf(&text,
                                      "Please enter correct PIN for security token '%s' in order to unlock %s (final try):",
                                      token_label, friendly_name);
+                        if (r < 0)
+                                return log_oom();
+                }
+
```

Comment:
hmm, nah, we jus missed the "else" here. i.e. we only should call one asprintf() here and initialize the `text` variable only once. i.e. the fix is much simpler, just `if` → `else if` in the next line

Reply:
I see, thank you, this definitely makes much more sense.